tests/core/celery/tasks/test_generate_inventory_and_hold_reports.py

```diff
@@ -0,0 +1,279 @@
+    # for testing, don't delete the files associated with the attachments so we can read them after the script
+    # runs
+    GenerateInventoryAndHoldsReportsJob._delete_attachments = False
+    job.services.email.container.send_email = send_email_mock
+    job.run()
+    send_email_mock.assert_called_once()
+    args, kwargs = send_email_mock.call_args
+    assert kwargs["receivers"] == [email]
+    assert "Inventory and Holds Reports" in kwargs["subject"]
+    attachments: dict = kwargs["attachments"]
+
+    assert len(attachments) == 2
+    inventory_report_key = [x for x in attachments.keys() if "inventory" in x][0]
+    assert inventory_report_key
+    assert "test_library" in inventory_report_key
+    inventory_report_value = attachments[inventory_report_key]
+    assert inventory_report_value
+    inventory_report_reader = csv.DictReader(open(inventory_report_value))
+    inventory_row_count = 0
+
+    for row in inventory_report_reader:
+        inventory_row_count += 1
+        assert row["title"] == title
+        assert row["author"] == author
+        assert row["identifier"]
+        assert row["language"] == language
+        assert row["publisher"] == publisher
+        assert row["audience"] == "young adult"
+        assert row["genres"] == "genre_a,genre_z"
+        assert row["format"] == edition.BOOK_MEDIUM
+        assert row["data_source"] == data_source
+        assert row["collection_name"] == collection_name
+        assert float(row["days_remaining_on_license"]) == float(days_remaining)
+        assert row["shared_active_loan_count"] == "0"
+        assert row["library_active_loan_count"] == "0"
+        assert row["remaining_loans"] == str(checkouts_left)
+        assert row["allowed_concurrent_users"] == str(terms_concurrency)
+        assert expiration.strftime("%Y-%m-%d %H:%M:%S.%f") in row["license_expiration"]
+
+    assert inventory_row_count == 1
+
+    holds_report_key = [x for x in attachments.keys() if "holds" in x][0]
+    assert holds_report_key
+    assert "test_library" in holds_report_key
+    holds_report_value = attachments[holds_report_key]
+    assert holds_report_value
+    holds_report_reader = csv.DictReader(open(holds_report_value))
+    row_count = 0
+
+    for row in holds_report_reader:
+        row_count += 1
+        assert row["title"] == title
+        assert row["author"] == author
+        assert row["identifier"]
+        assert row["language"] == language
+        assert row["publisher"] == publisher
+        assert row["audience"] == "young adult"
+        assert row["genres"] == "genre_a,genre_z"
+        assert row["format"] == edition.BOOK_MEDIUM
+        assert row["data_source"] == data_source
+        assert row["collection_name"] == collection_name
+        assert int(row["shared_active_hold_count"]) == shared_patrons_in_hold_queue
+        assert int(row["library_active_hold_count"]) == 3
+
+        assert row_count == 1
```

Comment:
Minor: This doesn't matter so much because there is only a single row here, but this line and the following lines should be out-dented so they are not being run in the `for` loop.